Tools/autotest/arducopter.py

```diff
@@ -14278,12 +14278,25 @@ def PLDNoParameters(self):
 
     def LuaParamSet(self):
         '''test param-set.lua applet'''
-        self.install_applet_script_context("param-set.lua")
         self.set_parameters({
             'SCR_ENABLE': 1,
         })
+
+        self.context_push()
+
+        self.install_applet_script_context("param-set.lua")
         self.reboot_sitl()
 
+        # collect original parameter values for everything other than
+        # SCR_ENABLE and PARAM_SET_ENABLE.  This resolves an issue at
+        # the end of the test where the popped context wants to set
+        # PARAM_SET_ENABLE back to its original value (enabled!),
+        # which can cause problems resetting
+        orig_parameter_values = self.get_parameters([
+            'RTL_ALT',
+            'DISARM_DELAY',
+        ])
```

Comment:
Concerned about this getting out of date.

Reply:
Not sure what you mean here; we need to restore these original parameter values before we context-pop so the context-pop works when we're in this weird state where we must have the script running so we can revert its parameter but don't want that parameter being set to its default enabled state - which can then prevent is from setting these two parameters!

Reply:
The problem is if the test gets changed to exercise different params for whatever reason, but this doesn’t get updated, it will still work but will introduce a race condition again.